lib/include/openamp/virtio_mmio_dev.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023, STMicroelectronics
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef OPENAMP_VIRTIO_MMIO_DEV_H
+#define OPENAMP_VIRTIO_MMIO_DEV_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <metal/utilities.h>
+#include <metal/device.h>
+#include <openamp/virtio_mmio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Define an Empty MMIO register table with only the strict necessary
+ * for a driver to recognise the device
+ *
+ * @note The initial state NOT_READY is the current approach to block the
+ * device side usage from driver until it gets properly configured.
+ */
+#define EMPTY_MMIO_TABLE {					\
+	.magic = VIRTIO_MMIO_MAGIC_VALUE_STRING,		\
+	.version = 2,						\
+	.status = VIRTIO_CONFIG_STATUS_NOT_READY,		\
+}
+
+/**
+ * @brief MMIO Device Registers: 256 bytes in practice, more if the configuration space is used.
+ * This is a trimmed down version with only the essential values needed to be detected correctly.
+ */
+struct mmio_table {
```

Comment:
It is the same case of EMPTY_MMIO_TABLE case, but I agree with you, when merging my work with ST I was in doubt what is the better way to handle the configuration device area, and I find this more scalable instead of direct addressing, but from MMIO perspective I think direct addressing might be a good idea, when I did the initial work I made this way:

https://github.com/uLipe/virtio-mmio-serial-rtos-to-rtos/blob/0c6c7e1d1e7dd4952ac1ef4a1b6c6e09473b70c6/device_side/src/virtio-mmio-backend.c#L32